src/dbos-executor.ts

```diff
@@ -647,6 +667,30 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }
     const wConfig = wInfo.config;
 
+    // Compatibility with the old way of calling workflows, which would include a parentCtx
+    const pctx = getCurrentContextStore();
+    const passContext = wInfo.registration?.passContext ?? true;
+    if (!passContext && pctx) {
+      const span = this.tracer.startSpan(
+        wf.name,
+        {
+          operationUUID: workflowUUID,
+          operationType: OperationType.WORKFLOW,
+          status: StatusString.PENDING,
+          authenticatedUser: pctx.authenticatedUser,
+          assumedRole: pctx.assumedRole,
+          authenticatedRoles: pctx.authenticatedRoles,
+        }
+        // TODO the span should be taken from pctx
+        //pctx.span
+      );
+      params.parentCtx = new DBOSContextImpl(wf.name, span, this.logger);
+      params.parentCtx.request = pctx.request || {};
+      params.parentCtx.authenticatedUser = pctx.authenticatedUser || "";
+      params.parentCtx.assumedRole = pctx.assumedRole || "";
+      params.parentCtx.authenticatedRoles = pctx.authenticatedRoles || [];
+      params.parentCtx.workflowUUID = workflowUUID;
+    }
```

Comment:
@chuck-dbos If the workflow is a child workflow, I think we'll not need to do this custom extraction. This is only useful when the workflow is a "root" workflow, `params.parentCtx` is not set, and we have to fill it manually to have a full `WorkflowContextImpl`. Can I just add a definition check on `params.parentCtx` in the `if` condition gating this logic?

Reply:
Didn't you (or copilot) write that?  I thought you knew ;-)